prometheus/promauto/auto_test.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package promauto
+
+import (
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func TestWrapNil(t *testing.T) {
+	// A nil registerer should be treated as a no-op by promauto, even when wrapped.
+	registerer := prometheus.WrapRegistererWith(prometheus.Labels{"foo": "bar"}, nil)
```

Comment:
I think test for Wrap should be in `prometheus` package and there is no need for using promauto there.

In the same time same test might be here for just `With` (:

Reply:
I wanted to use `promauto` as that is where the assumption that a `nil` registerer comes from, and I wanted to show that the `Wrap..` functions should also work.  I can add another test in `prometheus/` if you like?

Reply:
Yes, would be good to quickly test in the `prometheus` package that wrapping `nil` works.